### PR TITLE
[v1.3] Refactor room package and fix UI bugs in v1.2

### PR DESCRIPTION
**Pull request by adamwth**
### Main

- Changed all Room classes except UniqueRoomList, and Expenses to be immutable. Expenses will be done by @teowz46
- Refactored tests, like CheckoutCommandTest and RoomTest
- Refactored test utils, like TypicalRooms, TypicalBookings
- Fixed style in user guide
- UI not updating appropriately upon command in v1.2 was due to mutable changes to Room object not being detected by ObservableList, which is expected behavior of ObservableList. After refactoring Room (and the associated classes) to immutable, this bug has been fixed.
- Fixed commands_log in .git_ignore

### CODE REVIEW

**Comment by adamwth** [comment]
I have no idea why Travis is failing here. This branch passes on my local, using the same checks that Travis uses.

File: src/main/java/seedu/address/logic/CommandHistory.java [hunk 1]
```diff
@@ -41,7 +41,8 @@ public void add(String userInput) {
         }
         String inputString = string.toString();
         System.out.println(inputString);
-        CommandArchive.stringToFile(inputString);
+        // Fix this!!!
+        // CommandArchive.stringToFile(inputString);
```
Comment (pikulet) [inline src/main/java/seedu/address/logic/CommandHistory.java]: @neilish3re Please make an issue and implement this section

File: src/main/java/seedu/address/model/room/booking/Bookings.java [hunk 2]
```diff
@@ -3,201 +3,168 @@
...
-    public void setBookings(List<Booking> bookings) {
-        requireAllNonNull(bookings);
-        if (!bookingsAreNotOverlapping(bookings)) {
-            throw new OverlappingBookingException();
-        }
-
-        internalList.setAll(bookings);
+    public boolean canAcceptBooking(Booking toCheck) {
+        requireNonNull(toCheck);
+        return sortedBookingsSet.stream().noneMatch(toCheck::isOverlapping);
```
Comment (pikulet) [inline src/main/java/seedu/address/model/room/booking/Bookings.java]: Not a major issue, but this section can actually be done in parallel!

Comment (pikulet) [reply src/main/java/seedu/address/model/room/booking/Bookings.java]: Also applies to other methods which require iterating through many `Room`s or `Booking`s.

File: src/test/java/seedu/address/testutil/TypicalBookings.java [hunk 3]
```diff
@@ -10,7 +13,7 @@
     public static final Booking LASTWEEK_YESTERDAY = new BookingBuilder()
         .withBookingPeriod(TypicalBookingPeriods.LASTWEEK_YESTERDAY).build();
 
-    public static final Booking LASTWEEK_YESTERDAY_CHECKED_IN = new BookingBuilder()
+    public static final Booking LASTWEEK_YESTERDAY_CHECKEDIN = new BookingBuilder()
         .withBookingPeriod(TypicalBookingPeriods.LASTWEEK_YESTERDAY)
         .withCheckIn(true).build();
```
Comment (pikulet) [inline src/test/java/seedu/address/testutil/TypicalBookings.java]: Once you implement `checkout` command with the removal of `Guest`, then these `TypicalBooking`s would have to include different `Guest`s. 

**Note that `Guest` should not repeat across ANY bookings if `checkout` intends to remove the `Guest` without checking through all bookings first. E.g. Amy is booked today-tomorrow, we check her out and delete her. But, she still has a booking for next week.

tldr; Concierge currently does not support simultaneous bookings by the same `Guest`. This is caught by the `DuplicateGuestException` at adding booking.

File: src/test/java/seedu/address/testutil/TypicalRooms.java [hunk 4]
```diff
@@ -43,29 +41,22 @@
...
-            .withRoomNumber(TypicalRoomNumbers.ROOM_NUMBER_023.toString()).withCapacity(CAPACITY_SINGLE_ROOM)
-            .build();
+        .withRoomNumber(TypicalRoomNumbers.ROOM_NUMBER_023.toString()).withCapacity(CAPACITY_SINGLE_ROOM)
+        .build();
 
     public static final Room DOUBLE_024 = new RoomBuilder()
-            .withRoomNumber(TypicalRoomNumbers.ROOM_NUMBER_024.toString()).withCapacity(CAPACITY_DOUBLE_ROOM)
-            .build();
+        .withRoomNumber(TypicalRoomNumbers.ROOM_NUMBER_024.toString()).withCapacity(CAPACITY_DOUBLE_ROOM)
+        .build();
```
Comment (pikulet) [inline src/test/java/seedu/address/testutil/TypicalRooms.java]: Actually, indentation for wrapped lines should be 8 spaces.

https://oss-generic.github.io/process/codingStandards/CodingStandard-Java.html Layout point 2.

File: src/main/java/seedu/address/model/room/booking/Bookings.java (hunk 2 above)
Comment (adamwth) [reply src/main/java/seedu/address/model/room/booking/Bookings.java]: Okay noted. Same can be said for everywhere `stream` is used.

File: src/test/java/seedu/address/testutil/TypicalBookings.java (hunk 3 above)
Comment (adamwth) [reply src/test/java/seedu/address/testutil/TypicalBookings.java]: Yes, so `add` command should be refactored to not add the guest to the UniqueGuestList. Are you planning to do this?

File: src/test/java/seedu/address/testutil/TypicalRooms.java (hunk 4 above)
Comment (adamwth) [reply src/test/java/seedu/address/testutil/TypicalRooms.java]: Noted.
